src/wagtailtrans/models.py

```diff
@@ -202,109 +279,83 @@ def move_translated_pages(self, canonical_target, pos=None):
+    def get_translations(self, only_live=True, include_self=False):
+        """Get all translations of this page.
 
-        :param language: Language instance
-        :return: Boolean
+        This page itself is not included in the result, all pages
+        are sorted by the language position.
+
+        :param only_live: Boolean to filter on live pages & languages.
+        :return: Page instance
 
         """
-        return language.pages.filter(canonical_page=self).exists()
+        items = self.get_translatable_page_items(only_live=only_live, include_self=include_self).values_list('page__pk', flat=True)
+        translations = Page.objects.filter(pk__in=items)
+
+        return translations
 
     def get_translation_parent(self, language):
         site = self.get_site()
-        if not language.has_pages_in_site(site):
+
+        if not language.has_pages_in_site(site) or not self.language:
             return site.root_page
 
         translation_parent = (
-            TranslatablePage.objects
-            .filter(canonical_page=self.get_parent(), language=language, path__startswith=site.root_page.path)
+            TranslatablePageItem.objects.filter(
+                page__path__startswith=site.root_page.path,
+                canonical_page=self.get_parent(),
+                language=language,
+            )
             .first()
         )
-        return translation_parent
 
-    def create_translation(self, language, copy_fields=False, parent=None):
-        """Create a translation for this page. If tree syncing is enabled the
-        copy will also be moved to the corresponding language tree.
+        if translation_parent:
+            return translation_parent.page
 
-        :param language: Language instance
-        :param copy_fields: Boolean specifying if the content should be copied
-        :param parent: Parent page instance for the translation
-        :return: new Translated page (or subclass) instance
-
-        """
-        if self.has_translation(language):
-            raise Exception("Translation already exists")
-
-        if not parent:
-            parent = self.get_translation_parent(language)
-
-        if self.slug == self.language.code:
-            slug = language.code
-        else:
-            slug = '%s-%s' % (self.slug, language.code)
-
-        update_attrs = {
-            'title': self.title,
-            'slug': slug,
-            'language': language,
-            'live': False,
-            'canonical_page': self,
-        }
-
-        if copy_fields:
-            kwargs = {'update_attrs': update_attrs}
-            if parent != self.get_parent():
-                kwargs['to'] = parent
+        return None
 
-            new_page = self.copy(**kwargs)
-        else:
-            model_class = self.content_type.model_class()
-            new_page = model_class(**update_attrs)
-            parent.add_child(instance=new_page)
 
-        return new_page
+class TranslatablePage(Page):
```

Comment:
Would be great if we can remove this one and only use a mixin

Reply:
Yes we are 100% gonna remove this model. Left the code I didn't touch just to remind me we maybe need to do something with this (`translatable_page_ptr`).